### PR TITLE
Add co-locale documentation

### PR DESCRIPTION
**Pull request by jhh67**

### CODE REVIEW

File: doc/rst/usingchapel/multilocale.rst [hunk 1]
```diff
@@ -175,6 +175,37 @@ smp                  fast
...
+performance may be improved by running one locale in each socket to avoid
+inter-socket memory latencies. We refer to this functionality as *co-locales*
+and currently Chapel can run one locale per socket in the following
+configurations:
+
+=========  =================== =============
+CHPL_COMM  CHPL_COMM_SUBSTRATE CHPL_LAUNCHER
+=========  =================== =============
+ofi        NA                  slurm-srun
+gasnet     ibv                 slurm-srun
```
[the remark this answers is not in the document]
Comment (jhh67) [reply doc/rst/usingchapel/multilocale.rst:193]: I have it working for slurm-srun but not slurm-gasnetrun_ibv (yet).

[the remark this answers is not in the document]
Comment (ronawho) [reply doc/rst/usingchapel/multilocale.rst:193]: I see. I don't think we document how to build gasnet-ibv with slurm-srun support (usually need to provide explicit paths to pmi.) But maybe that's ok for now.

**Comment by ronawho** [comment]
I think it'd be worth adding some wiggle words like to note co-locales is a prototype, experimental, nascent, or whatever. Just something to give people a heads up that it is early days for it.

**Comment by jhh67** [comment]
> I think it'd be worth adding some wiggle words like to note co-locales is a prototype, experimental, nascent, or whatever. Just something to give people a heads up that it is early days for it.

Done.

**Review by ronawho — APPROVED** [review]

File: doc/rst/usingchapel/multilocale.rst (hunk 1 above)
[the remark this answers is not in the document]
Comment (jhh67) [reply doc/rst/usingchapel/multilocale.rst:193]: No problem, I should have linked #23964 to #23467, which I've now done.

**Pull request merged**